docs/development_environment.mdx

```diff
@@ -44,6 +44,16 @@ The devcontainer comes with some useful tasks to help you with development, you
 
 When a task is currently running (like `Preview` for the docs), it can be restarted by opening the command palette and selecting `Tasks: Restart Running Task`, then select the task you want to restart.
 
+### Debugging in Visual Studio Code
+If you need to debug an integration or Home Assistance Core you need to ensure the following:
+1. You have the `debugpy` component enabled in `configuration.yaml`. More on this [here](https://www.home-assistant.io/integrations/debugpy/)
+2. You use the right debug configuration in Visual Studio Code defined in your `launch.json` file. Sample configurations are described in the `debugpy` integration page. To debug code residing in VS Code Dev Container you should use the `Python: Attach Local` config
+
+After making sure you met the above prerequisites - you need to start the debugger, which will try to connect to a Home Assistant running instance on the port indicated in your `launch.json` config file.
+A successfull debugging setup will be indicated by Visual Studio Code by changing the status bar at the bottom to orange with a note of the debugging configuration used (e.g. `Python: Attach Local`).
+
+If you followed the above, the breakpoints you put in Visual Studio Code should work and the debugger will stop when reaching a breakpoint. 
+
```

Comment:
[Here ](https://github.com/home-assistant/developers.home-assistant/blob/2e61d9d5ca92786c83590e0e9a8798587786601f/docs/development_environment.mdx?plain=1#L43)the instructions explain about dev container, if you followed these instructions and debugging your local instance these instructions about debugpy are not needed and might confuse new developers.
I think it should be clear that the instructions above are for debugging a remote instance (as also explained in the debugpy integration page) and not for developing inside the dev container.

Reply:
I'm not sure what you mean by pointing to the 'Tasks' section. There is no task to enable debugging in the dev container. 

I put this section about debugging within the 'Dev Container' section, so I think it won't confuse anyone going through the devcontainer guide. 

Debugging within devcontainer is only possible using the debugpy integration. It doesn't work out-of-the-box.

Reply:
> Debugging within devcontainer is only possible using the debugpy integration. It doesn't work out-of-the-box.
It does work out of the box if you debug the local instance, it might be that your dev environment is not setup correctly and it would be incorrect to point users to do unnecessary steps for that.
The line I linked was only to indicate this article explains about dev container. For debugging your dev container you don't need to do anything but start debug:
![image](https://user-images.githubusercontent.com/1858925/190176131-2cb62894-cc5e-4c8e-8abb-5d499ab0bbce.png)

Reply:
Hmm, I've setup my dev container 3 days ago using the guide and lost almost 2 hours trying to get the debugging to work. Since the docs do not mention anything about it - I thought it's just not covered.
The only way I've found to debug my local dev container instance is using the `debugpy`.

I still think it makes sense to mention debugging in the docs as it looks like it's not covered in there and may confuse the ones, who start with HA development. Especially in case of issues.

I'm not sure why it doesn't work for me and it does for you, but I think it's worth to mention this solution for the remote debuggin and a workaround for issues with dev container local debugging (like in my case).

BTW. What does your `launch.json` file contain when you start debugging?? I guess there's still some configuration used for that. Perhaps if I change something  in it I will get it working. But now after I press the F5 button with the 'Home Assistant' config selected I'm getting this:
![image](https://user-images.githubusercontent.com/27864579/190178815-d799390c-081e-4322-b71a-80a93b9cc60b.png)

Reply:
I've used some solutions from the web and I got it working. Not sure how stable these solutions are and if it's always gonna work for me, but for now every time when I hit F5 I'm getting HA opened by VS Code integrated debugpy lancher.

I'm gonna close this PR and will open another one and briefly decsribe the debugging abilities (that they are enabled by default, and remore debugging is possible using debugpy).

Reply:
Thanks, I highly recommend to join our [Discord chat server](https://discord.gg/c5DvZ4e), you can get quick answers for problems like this directly from members or developers so you don't have to fight with it when it is not working.

Reply:
> Thanks, I highly recommend to join our [Discord chat server](https://discord.gg/c5DvZ4e), you can get quick answers for problems like this directly from members or developers so you don't have to fight with it when it is not working.

This must have been some untypical issue as noone from the Discord chat was able to help or advise. But I'm glad I've found a working solution.